src/core/net/dns_headers.hpp

```diff
@@ -493,6 +493,138 @@ class Name
+     * This constructor initializes the `Name` object with a given string.
+     *
+     * @param[in] aString   A string specifying the name (dot '.' separated sequence of labels').
+     *
+     */
+    explicit Name(const char *aString)
+        : Name(aString, nullptr, 0)
+    {
+    }
+
+    /**
+     * This constructor initializes the `Name` object from a message at a given offset.
+     *
+     * @param[in] aMessage   The message containing the encoded name. `aMessage.GetOffset()` MUST point to the start of
+     *                       the DNS header in the message (used to parse compressed name)..
+     * @param[in] aOffset    The offset in @p aMessage pointing to the start of the name.
+     *
+     */
+    Name(const Message &aMessage, uint16_t aOffset)
+        : Name(nullptr, &aMessage, aOffset)
+    {
+    }
+
+    /**
+     * This method indicates whether the name is empty (not specified).
+     *
+     * @returns TRUE if the name is empty, FALSE otherwise.
+     *
+     */
+    bool IsEmpty(void) const { return (mString == nullptr) && (mMessage == nullptr); }
+
+    /**
+     * This method indicates whether the name is specified from a string.
+     *
+     * @returns TRUE if the name is specified from a string, FALSE otherwise.
+     *
+     */
+    bool IsFromString(void) const { return mString != nullptr; }
+
+    /**
+     * This method indicates whether the name is specified from a message.
+     *
+     * @returns TRUE if the name is specified from a message, FALSE otherwise.
+     *
+     */
+    bool IsFromMessage(void) const { return mMessage != nullptr; }
+
+    /**
+     * This method gets the type of `Name` object indicating whether it is empty, specified by a string or from a
+     * message
+     *
+     * @returns The name type.
+     *
+     */
+    Type GetType(void) const { return IsFromString() ? kTypeString : (IsFromMessage() ? kTypeMessage : kTypeEmpty); }
+
+    /**
+     * This method sets the name from a given string.
+     *
+     * @param[in] aString   A string specifying the name (dot '.' separated sequence of labels').
+     *
+     */
+    void SetFromString(const char *aString)
+    {
+        mString  = aString;
+        mMessage = nullptr;
+    }
+
+    /**
+     * This method sets the name from a message at a given offset.
+     *
+     * @param[in] aMessage   The message containing the encoded name. `aMessage.GetOffset()` MUST point to the start of
+     *                       the DNS header in the message (used to parse compressed name)..
+     * @param[in] aOffset    The offset in @p aMessage pointing to the start of the name.
+     *
+     */
+    void SetFromMessage(const Message &aMessage, uint16_t aOffset)
+    {
+        mString  = nullptr;
+        mMessage = &aMessage;
+        mOffset  = aOffset;
+    }
+
+    /**
+     * This method gets the name as string.
+     *
+     * This method MUST be used only when the type if `kTypeString` and `IsFromString()` returns TRUE. Otherwise its
```

Comment:
```suggestion
     * This method MUST be used only when the type is `kTypeString` and `IsFromString()` returns TRUE. Otherwise its
```

__the type is `kTypeString`__  and __`IsFromString()` returns TRUE__ seem equivalent, is it necessary to repeat here?